Updated README (#15)
cyber-crypt-com/encryptonize-core: README.md

data across multiple cloud environments while simultaneously removing the need to trust the cloud
providers with encryption keys. Encryptonize Core provides a simple gRPC interface that allows multiple workloads or users to encrypt and decrypt data securely.

Encryptonize Core is designed with the philosophy that you should not have to unconditionally trust cloud
providers to protect your sensitive data, and that having to manage keys and credentials across
several cloud environments is a security problem in itself. Encryptonize Core minimizes the
potential attack surface and simplifies key management, while still allowing you to take advantage
of a multi cloud setup. It is designed to not only encrypt and authenticate your data using modern
cryptographic standards, but also to cryptographically enforce user authentication and
authorization.

## Running Encryptonize&reg; Core locally

The easiest way to try out Encryptonize Core is to start a local Docker Compose instance. To do
this, run

```bash
make docker-up
```

This will start a complete Encryptonize Core instance inside Docker. To shut down the
Encryptonize Core instance, call

```bash
make docker-down
```

> For more information about how to get started with Encryptonize Core, see [documentation/getting_started.md](documentation/getting_started.md). The documentation folder also provides a [documentation/user_manual.md](documentation/user_manual.md) as well as an overview of the [documentation/api.md](documentation/api.md).

Reviewer: ```suggestion
For more information about how to get started with Encryptonize Core, see the [Getting Started guide](documentation/getting_started.md). The documentation folder also provides a [User Manual](documentation/user_manual.md) as well as an overview of the [gRPC API](documentation/api.md).
```

### Deployment

Using the Encryptonize Core Helm Chart, you can deploy Encryptonize Core to a Kubernetes.

See the Encryptonize Core Helm Chart [README](deployment/charts/README.md) for more information.

## Running Encryptonize&reg; Core manually

Encryptonize Core can also be run manually using the gRPC API. Note that this is only recommended for development. To do this, build the executable:

```bash

> ./encryptonize-core

Author: Fixed

```

> Afterwards, the gRPC service endpoints can be called with e.g. the tool [gRPCurl](https://github.com/fullstorydev/grpcurl). See the [documentation/user_manual.md](https://github.com/cyber-crypt-com/encryptonize-core/blob/master/documentation/user_manual.md) for more information.

Reviewer: ```suggestion
Afterwards, the gRPC service endpoints can be called with e.g. the tool [gRPCurl](https://github.com/fullstorydev/grpcurl). See the [User Manual](https://github.com/cyber-crypt-com/encryptonize-core/blob/master/documentation/user_manual.md) for more information.
```

### Deployment

Using the Encryptonize Core Helm Chart, you can deploy Encryptonize Core to a Kubernetes.

See the Encryptonize Core Helm Chart [README](deployment/charts/README.md) for more information.

## Running Encryptonize&reg; Core manually

> Encryptonize Core can also be run manually using the gRPC API. Note that this is only recommended for development. To do this, run the executable:

Reviewer: ```suggestion
Encryptonize Core can also be run manually. Note that this is only recommended for development. To do this, run the executable:
```

```bash
./encryptonize-core
```

Afterwards, the gRPC service endpoints can be called with e.g. the tool [gRPCurl](https://github.com/fullstorydev/grpcurl). See the [documentation/user_manual.md](https://github.com/cyber-crypt-com/encryptonize-core/blob/master/documentation/user_manual.md) for more information.

## Repository overview

```

> ├── .github
> │   └── workflows                   # CI/CD files
> ├── common
> ├── config
> ├── deployment                  # Deployment information
> ├── documentation               # User documentation
> │   ├── api.md                      # API specifications
> │   ├── getting_started.md          # Getting started guide
> │   └── user_manual.md              # User manual
> ├── e2etests                    # End-to-end tests
> ├── protobuf                    # Protobuf files
> ├── service                     # Source files for Encryptonize Core
> ├── storage                     # Storage files
> ├── .dockerignore
> ├── Dockerfile                  
> ├── README.md                   # This file
> ├── config.toml                 # Example configuration file
> ├── docker-compose.yml          
> ├── go.mod
> ├── go.sum
> ├── main.go
> └── makefile

Reviewer: I think we should just explain the folders that are relevant, e.g.:
```suggestion
├── deployment                  # Deployment information
├── documentation               # User documentation
│   ├── api.md                      # API specifications
│   ├── getting_started.md          # Getting started guide
│   └── user_manual.md              # User manual
├── e2etests                    # End-to-end tests
├── protobuf                    # Protobuf files
├── service                     # Source files for Encryptonize Core
├── Dockerfile                  # Encryptonize Core Docker image definition
├── README.md                   # This file
├── config.toml                 # Example configuration file
├── docker-compose.yml          # Example Docker Compose setup
```
